tests/integration/ef_tests/runner.py

```diff
@@ -0,0 +1,130 @@
+from starkware.starknet.testing.contract import DeclaredClass, StarknetContract
+from state_management import (
+    fund_and_set_allowance,
+    write_test_state_for_contract,
+    write_test_state_for_eoa,
+)
+
+from tests.utils.helpers import hex_string_to_bytes_array
+from tests.utils.uint256 import int_to_uint256
+from utils import display_storage, is_account_eoa
+
+
+async def write_test_state(
+    account_proxy_class: DeclaredClass,
+    contract_account_class: DeclaredClass,
+    externally_owned_account_class: DeclaredClass,
+    get_contract_account,
+    get_starknet_address,
+    eth: StarknetContract,
+    kakarot: StarknetContract,
+    starknet: StarknetContract,
+    pre_state,
+):
+    """
+    Initialize the Starknet state using an ef-test BlockchainTest format case 'pre' field.
+
+    See: https://ethereum-tests.readthedocs.io/en/latest/test_filler/blockchain_filler.html#pre for details.
+    """
+    for address, state in pre_state.items():
+        evm_address = int(address, 16)
+        starknet_address = get_starknet_address(evm_address)
+
+        balance = int_to_uint256(int(state["balance"], 16))
+        await fund_and_set_allowance(
+            eth, kakarot.contract_address, starknet_address, balance
+        )
+
+        if is_account_eoa(state):
+            await write_test_state_for_eoa(
+                externally_owned_account_class.class_hash,
+                account_proxy_class.class_hash,
+                kakarot,
+                starknet,
+                state,
+                evm_address,
+                starknet_address,
+            )
+        else:
+            contract = get_contract_account(starknet_address)
+            await write_test_state_for_contract(
+                contract_account_class.class_hash,
+                account_proxy_class.class_hash,
+                contract,
+                kakarot,
+                starknet,
+                state,
+                evm_address,
+                starknet_address,
+            )
+
+
+async def do_transaction(
+    blocks,
+    get_starknet_address,
+    kakarot: StarknetContract,
+    owner,
+    starknet: StarknetContract,
+):
+    """
+    Execute each transaction in each block using an ef-test BlockchainTest format's `blocks` field.
+
+    See: https://ethereum-tests.readthedocs.io/en/latest/test_filler/blockchain_filler.html#blocks for details.
+    """
+    for block in blocks:
+        transactions = block["transactions"]
+
+        for transaction in transactions:
+            starknet_address = get_starknet_address(int(transaction["sender"], 16))
+
+            await kakarot.eth_send_transaction(
```

Comment:
this is currently different than how the rust runner handles transactions. per a discussion, it may be worth making sure they are ran the same way where our entry point is through an eoa that then calls into this. open for feedback here.

Reply:
a concrete case that is failing on the ef-tests rust runner and this runner that I think is because of the different entry point is `CallContractToCreateContractNoCash_d0g0v0_Shanghai`